src/shared/ui/solid-button/SolidButton.tsx

```diff
@@ -0,0 +1,71 @@
+'use client';
+
+import type { ComponentProps, ButtonHTMLAttributes } from 'react';
+import { forwardRef } from 'react';
+import { SurfIcon } from '@/shared/ui/icon/SurfIcon';
+
+type SurfIconName = ComponentProps<typeof SurfIcon>['name'];
+
+type ButtonSize = 's' | 'm' | 'l';
+type ButtonVariant = 'primary' | 'secondary' | 'danger' | 'warning';
+
+type SolidButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  size: ButtonSize;
+  variant: ButtonVariant;
+  leftIconName?: SurfIconName | null;
+  rightIconName?: SurfIconName | null;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+export const SolidButton = forwardRef<HTMLButtonElement, SolidButtonProps>(function SolidButton({
+  size = 'm',
+  variant = 'primary',
+  disabled = false,
+  leftIconName,
+  rightIconName,
+  children,
+  type = 'button',
+  onClick,
+  ...rest
+}: SolidButtonProps) {
```

Comment:
@rngrhn4114
<img width="1168" height="609" alt="image" src="https://github.com/user-attachments/assets/9f5bdd90-93e6-4346-94be-30975b23d372" />
```suggestion
  ...rest
}, ref) {
```